plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -932,10 +952,38 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
 
         try {
             TrinoFileSystem fileSystem = fileSystemFactory.create(session);
-            if (!fileSystem.listFiles(deltaLogDirectory).hasNext()) {
+            boolean transactionLogFileExists = fileSystem.listFiles(deltaLogDirectory).hasNext();
+
+            if (!replaceExistingTable && transactionLogFileExists) {
+                throw new TrinoException(
+                        NOT_SUPPORTED,
+                        "Using CREATE TABLE with an existing table content is disallowed, instead use the system.register_table() procedure.");
+            }
+            else {
+                long commitVersion = 0;
                 TransactionLogWriter transactionLogWriter = transactionLogWriterFactory.newWriterWithoutTransactionIsolation(session, location);
+                List<AddFileEntry> activeFiles = ImmutableList.of();
+                ProtocolEntry protocolEntry;
+                if (replaceExistingTable) {
+                    commitVersion = getMandatoryCurrentVersion(fileSystem, location) + 1;
+                    transactionLogWriter = transactionLogWriterFactory.newWriter(session, location);
+                    activeFiles = transactionLogAccess.getActiveFiles(
+                            getSnapshot(session, tableHandle),
+                            tableHandle.getMetadataEntry(),
+                            tableHandle.getProtocolEntry(),
+                            tableHandle.getEnforcedPartitionConstraint(),
+                            tableHandle.getProjectedColumns(),
+                            session);
+                    protocolEntry = protocolEntryForTable(tableHandle.getProtocolEntry(), containsTimestampType, tableMetadata.getProperties());
+                    statisticsAccess.deleteExtendedStatistics(session, schemaTableName, location);
```

Comment:
Add a test covering the fact that the extended statistics of the table are missing after `CREATE OR REPLACE` table.

Maybe use `extended_statistics_collect_on_write` session property on `CREATE OR REPLACE TABLE AS SELECT` to showcase that the old statistics have been removed.